src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -274,27 +248,29 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
         [System.Diagnostics.Conditional("_DEBUG")]
         private void VerifyClassInvariant()
         {
-            Debug.Assert((uint)(m_ChunkOffset + m_ChunkChars.Length) >= m_ChunkOffset, "Integer Overflow");
+            // TODO: This assert looks like it will always succeed bc both sides are being converted to long.
+            Debug.Assert((uint)(m_ChunkOffset + m_ChunkChars.Length) >= m_ChunkOffset);
+            
             StringBuilder currentBlock = this;
             int maxCapacity = this.m_MaxCapacity;
             for (;;)
             {
                 // All blocks have copy of the maxCapacity.
-                Debug.Assert(currentBlock.m_MaxCapacity == maxCapacity, "Bad maxCapacity");
-                Debug.Assert(currentBlock.m_ChunkChars != null, "Empty Buffer");
+                Debug.Assert(currentBlock.m_MaxCapacity == maxCapacity);
+                Debug.Assert(currentBlock.m_ChunkChars != null);
 
-                Debug.Assert(currentBlock.m_ChunkLength <= currentBlock.m_ChunkChars.Length, "Out of range length");
-                Debug.Assert(currentBlock.m_ChunkLength >= 0, "Negative length");
-                Debug.Assert(currentBlock.m_ChunkOffset >= 0, "Negative offset");
+                Debug.Assert(currentBlock.m_ChunkLength <= currentBlock.m_ChunkChars.Length);
+                Debug.Assert(currentBlock.m_ChunkLength >= 0);
+                Debug.Assert(currentBlock.m_ChunkOffset >= 0);
 
                 StringBuilder prevBlock = currentBlock.m_ChunkPrevious;
                 if (prevBlock == null)
                 {
-                    Debug.Assert(currentBlock.m_ChunkOffset == 0, "First chunk's offset is not 0");
+                    Debug.Assert(currentBlock.m_ChunkOffset == 0);
```

Comment:
@danmosemsft, there is a `break` statement after the assert because this means we've reached a tail node. Without the `break` we would enter an infinite loop.